packages/walletconnect-core/src/connector.js

```diff
@@ -446,11 +446,10 @@ export default class Connector {
     let payload = this.checkObject(data, 'payload')
 
     if (payload.id) {
-      delete payload.id
```

Comment:
Actually we should just remove the `payload.id` check and always override the `payload.id`. Otherwise if the `payload.id` is missing, we end up with no ID at all